### PR TITLE
[WIP] Add a dpl.io standard definition

### PR DESCRIPTION
**Pull request by chrismaddern**

### CODE REVIEW

File: STANDARD.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+### Unencoded Query Parameters
+Basic dpl.io functionality is available through unencoded query parameters as defined below. These include definition of the standard version as well as the callback URL and attribution token.
+
+- `dpl:callback-url`: A URL to be considered as the 'back' button action
+- `dpl:protocol-version`: The version of the dpl.io protocol used to encode the deep link
+- `dpl:attribution-token`: A token to consider as the application's 'last touch' referrer. (More in Attribution & Referrer)
+
+### Encoding Complex Objects
+The dpl.io standard allows for the passing of complex objects between apps by JSON-encoding them and using a defined field to mark up encoded fields.
```
Comment (getaaron) [inline STANDARD.md]: "for the passing of complex objects" -> "complex objects to be passed"

File: STANDARD.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+The `dpl:json-encoded-fields` query parameter defines any other query parameters of the deep link that should be JSON-decoded and made available to the applciation as complex types.
```
Comment (getaaron) [inline STANDARD.md]: applciation -> application

File: STANDARD.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+By default, the following fields should be JSON-decoded:
+- `dpl:json-encoded-fields`: This field itself is an encoded array and should always be decoded first
+- `dpl:rerral-data`: This contains a dictionary of referrer information defined in Full Referrer Data
```
Comment (getaaron) [inline STANDARD.md]: `dpl:rerral-data` -> `dpl:referral-data`

File: STANDARD.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### Full Referrer Data
+The dictionary in the query parameter `dpl:referral-data` contains more granular information about the referring application. Specifically:
+- `dpl:referrer-application-itunes-id`: The iTunes ID of the referring application
+- - `dpl:referrer-application-play-id`: The iTunes ID of the referring application
```
Comment (getaaron) [inline STANDARD.md]: Extra "-" here. And did you mean Google Play ID?

File: STANDARD.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+- `dpl:referrer-application-name`: The display name of the referring application
+- `dpl:referrer-callback-display-title`: The title of the action button that invokes the URL in the `dpl:callback-url` query parameter.
```
Comment (getaaron) [inline STANDARD.md]: Has a period at the end but the other bullet points don't

**Comment by getaaron** [comment]
This looks pretty cool. For security, should any type of service be available to verify the information? (For example, someone might maliciously provide someone else's `dpl:referrer-application-itunes-id` but a `dpl:callback-url` with a scheme that doesn't match.)

**Comment by chrismaddern** [comment]
@getaaron Thanks for the feedback man! I'll go through and resolve them! I'd also love any thoughts on other basic requirements in deep linking which may make the cut for v1

**Pull request closed without merging**